sky/check.py

```diff
@@ -79,23 +100,39 @@ def check_one_cloud(
     disabled_clouds_set = {
         cloud for cloud in disabled_clouds if not cloud.startswith('Cloudflare')
     }
+    config_allowed_clouds_set = {
+        cloud for cloud in config_allowed_cloud_names
+        if not cloud.startswith('Cloudflare')
+    }
     previously_enabled_clouds_set = {
         repr(cloud) for cloud in global_user_state.get_cached_enabled_clouds()
     }
 
-    # Determine the set of enabled clouds: previously enabled clouds + newly
-    # enabled clouds - newly disabled clouds.
-    all_enabled_clouds = ((previously_enabled_clouds_set | enabled_clouds_set) -
-                          disabled_clouds_set)
+    # Determine the set of enabled clouds: (previously enabled clouds + newly
+    # enabled clouds - newly disabled clouds) intersected with
+    # config_allowed_clouds, if specified in config.yaml.
+    # This means that if a cloud is already enabled and is not included in
+    # allowed_clouds in config.yaml, it will be disabled.
+    all_enabled_clouds = (config_allowed_clouds_set & (
+        (previously_enabled_clouds_set | enabled_clouds_set) -
+        disabled_clouds_set))
     global_user_state.set_enabled_clouds(list(all_enabled_clouds))
 
+    skipped_clouds_hint = None
+    if skipped_clouds:
+        skipped_clouds_hint = (
+            '\nNote: The following clouds were disabled because they were not '
```

Comment:
Hmm do you mean even when `allowed_clouds` is not specified in config.yaml, `sky check <cloud>` should always show the hint above? 

I think in that case hint should not be shown because those clouds were not disabled because they were not in allowed_list, but rather because of other errors while checking (which are surfaced as `<cloud>: disabled ` along with reason). wdyt?

Reply:
oh, I mean when the `allowed_clouds` is specified and `sky check <cloud>` is used. It seems there is no hint showing in that case?

Reply:
I have a `~/.sky/config.yaml`, and when I do `sky check gcp` the following shows up, without the hint:
```yaml
allowed_clouds: ['aws', 'gcp', 'cloudflare']
```

```
  GCP: enabled                              

To enable a cloud, follow the hints above and rerun: sky check 
If any problems remain, refer to detailed docs at: https://skypilot.readthedocs.io/en/latest/getting-started/installation.html

🎉 Enabled clouds 🎉
  ✔ AWS
  ✔ GCP
```

Reply:
This is a nit which should not block us merging the PR :)

Reply:
Ahh got it! It should be fixed now :)